OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -85,32 +86,30 @@ public static Func<int> TurretFacingFromInit(IActorInitializer init, int def, st
 		}
 
 		public Turreted(ActorInitializer init, TurretedInfo info)
+			: base(info)
 		{
 			this.info = info;
 			TurretFacing = TurretFacingFromInit(init, info.InitialFacing, info.Turret)();
 		}
 
 		public void Created(Actor self)
 		{
-			attack = self.TraitOrDefault<AttackTurreted>();
+			attack = self.TraitsImplementing<AttackTurreted>().ToArray();
```

Comment:
Currently we support N-to-N AttackTurreted-Turreted linkings. As long as that is kept, Turreted.Tick() needs to stay as-is and so does this line.